extensions/notebook/src/test/book/book.test.ts

```diff
@@ -12,91 +12,140 @@ import * as rimraf from 'rimraf';
 import { BookTreeItem } from '../../book/bookTreeItem';
-import * as assert from 'assert';
 
 const SEED = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
 
+export interface ExpectedBookItem {
+	title: string;
+	url?: string;
+	sections?: any[];
+	external?: boolean;
+	previousUri?: string | undefined;
+	nextUri?: string | undefined;
+}
+
 describe('BookTreeViewProvider.getChildren', function (): void {
 	let rootFolderPath: string;
-	const expectedNotebook = {
-		title: 'Notebook',
-		url: '/notebook'
-	};
-	const expectedMarkdown = {
-		title: 'Markdown',
-		url: '/markdown'
-	};
-	const expectedExternalLink = {
-		title: 'GitHub',
-		url: 'https://github.com/',
-		external: true
-	};
-	const expectedBook = {
-		sections: [expectedNotebook, expectedMarkdown, expectedExternalLink],
-		title: 'Test Book'
-	};
+	let expectedNotebook1: ExpectedBookItem;
+	let expectedNotebook2: ExpectedBookItem;
+	let expectedNotebook3: ExpectedBookItem;
+	let expectedMarkdown: ExpectedBookItem;
+	let expectedExternalLink: ExpectedBookItem;
+	let expectedBook: ExpectedBookItem;
 
 	let mockExtensionContext: TypeMoq.IMock<vscode.ExtensionContext>;
 	let bookTreeViewProvider: BookTreeViewProvider;
 	let book: BookTreeItem;
+	let notebook1: BookTreeItem;
 
 	this.beforeAll(async () => {
-		try {
-			let testFolder = '';
-			for (let i = 0; i < 8; i++) {
-				testFolder += SEED.charAt(Math.floor(Math.random() * SEED.length));
-			}
-			rootFolderPath =  path.join(os.tmpdir(), 'BookTestData_' + testFolder);
-			let dataFolderPath = path.join(rootFolderPath, '_data');
-			let contentFolderPath = path.join(rootFolderPath, 'content');
-			let configFile = path.join(rootFolderPath, '_config.yml');
-			let tableOfContentsFile = path.join(dataFolderPath, 'toc.yml');
-			let notebookFile = path.join(contentFolderPath, 'notebook.ipynb');
-			let markdownFile = path.join(contentFolderPath, 'markdown.md');
-			await fs.mkdir(rootFolderPath);
-			await fs.mkdir(dataFolderPath);
-			await fs.mkdir(contentFolderPath);
-			await fs.writeFile(configFile, 'title: Test Book');
-			await fs.writeFile(tableOfContentsFile, '- title: Notebook\n  url: /notebook\n- title: Markdown\n  url: /markdown\n- title: GitHub\n  url: https://github.com/\n  external: true');
-			await fs.writeFile(notebookFile, '');
-			await fs.writeFile(markdownFile, '');
-			mockExtensionContext = TypeMoq.Mock.ofType<vscode.ExtensionContext>();
-			let folder: vscode.WorkspaceFolder = {
-				uri: vscode.Uri.file(rootFolderPath),
-				name: '',
-				index: 0
-			};
-			bookTreeViewProvider = new BookTreeViewProvider([folder], mockExtensionContext.object);
-		} catch (e) {
-			assert.ok(false, e);
+		let testFolder = '';
+		for (let i = 0; i < 8; i++) {
+			testFolder += SEED.charAt(Math.floor(Math.random() * SEED.length));
+		}
+		rootFolderPath = path.join(os.tmpdir(), 'BookTestData_' + testFolder);
+		if (os.type() === 'Windows_NT') {
+			rootFolderPath = rootFolderPath.replace('C:', 'c:');
```

Comment:
instead of doing this, let’s use a .tolower() when doing the comparison below

Reply:
Everywhere else, we use `if (process.platform === 'win32') {`. Can we be consistent here with our other test code?